nvaccess/nvda
# Switch to focus mode if read-only list items and their list parents are focusable and gain focus

### Link to issue number:

Fixes #13221

### Summary of the issue:

In the Microsoft Edge Downloads manager (CTRL+J), as well as in the channel and DM outputs of the Slack app, HTML lists and list items are used to display the items. These are made focusable via either aria-activedescendant, or the rowing tabIndex pattern. The reason is that for compatibility, these can be used to host rich content, such as whole sub documents or controls. Until now, NVDA would not automatically switch to focus mode when focus lands on such a focusable list item.

### Description of how this pull request fixes the issue:

This PR adds a rule for enabling passthrough mode if a list item that is read-only, has state focused, and at the same time its paarent is of role list and is focusable.

### Testing strategy:

1. Run NVDA.
2. Go into Microsoft Edge.
3. Press Control+J to invoke Downloads Manager.
4. Focus should be on the first downloaded item, and passthrough mode should automatically be enabled.

### Known issues with pull request:

None known.

### Change log entries:
New features
Changes
Bug fixes

* In the Microsoft Edge downloads manager, NVDA will now automatically switch to focus mode once the list item with the most recent download gains focus. #13221 

For Developers

### Code Review Checklist:



  - description is up to date
  - change log entries
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [ ] Documentation: Not needed since expected behavior is introduced where user would previously have to manually switch pass-through mode on.
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
